content/300-guides/200-deployment/110-deployment-guides/500-deploying-to-netlify.mdx

```diff
@@ -79,72 +72,97 @@ npm install
 ```terminal
 ls -1
 README.md
-functions
+functions/
 netlify.toml
-node_modules
+node_modules/
 package.json
-prisma
-public
+prisma/
+public/
 schema.sql
 ```
 
-## 3. Set the DATABASE_URL environment variable locally
+## 3. Get the Heroku PostgreSQL connection string
+
+You need the Heroku PostgreSQL connection string to define the necessary Prisma connection variable.
+
+1. In Heroku, open your app and select the **Resources** tab.
+2. Click the **Heroku Postgres** database add-on.
+3. Select the **Settings** tab in the page of the Heroku Postgres add-on.
+4. Click **View Credentials**.
+5. From the **URI** field, copy the entire connection string.
 
-Set the `DATABASE_URL` environment variable locally so you can create the database schema and Prisma can access the database to introspect:
+**Checkpoint:** The Heroku connection string has the following format:
+
+```
+postgres://__USER__:__PASSWORD__@__HOST__/__DATABASE__
+```
+
+When you next set the `DATABASE_URL` environment variable, you need to change the beginning string from `postgres` to `postgresql`.
+
+## 4. Set the DATABASE_URL environment variable locally
+
+Set the `DATABASE_URL` environment variable locally. The `prisma.schema` file uses this variable so that Prisma can access the database to introspect:
 
 ```terminal
 export DATABASE_URL="postgresql://__USER__:__PASSWORD__@__HOST__/__DATABASE__"
 ```
 
-> **Note:** you will need the `DATABASE_URL` environment variable for the subsequent steps. Set it in all terminal sessions related to this project.
+> **Note:** You need the `DATABASE_URL` environment variable for the subsequent steps. Set it in all terminal sessions related to this project.
 
 You need to replace the uppercase placeholders with your database credentials, e.g.:
 
 ```terminal
 postgresql://janedoe:randompassword@yourpostgres.compute-1.amazonaws.com:5432/yourdbname
 ```
 
-## 4. Create the database schema
+Or simply copy and paste the connection string you get from your cloud database provider.
 
-To create your database schema, run the `schema.sql` from the example code as follows:
+## 5. Create a shadow database in your cloud provider and get the connection string
```

Comment:
This is only true for some providers, and distract from the Netlify steps itself. It would fit into the suggested, optional "If you want to use Heroku for the database..." section that I mentioned above.

Reply:
Does it make sense to list the providers that require the manual creation of a shadow db? It would come with the advantage of giving our readers indication that if they're using provider X, Y, or Z, they need to go through these manual steps.

As a disadvantage, it means that we'd need to maintain this information in the future with updating the list with:
a) new providers
b) removing providers that no longer require this

Also, there's the disadvantage that we'd never have a complete list of providers.

Still, I'd feel safer as a reader and user of Prisma that Prisma is taking care of testing and fully informing me of this before I hit "Error P3104" and wonder what I need to do next and frustrate me in the process.

Reply:
Ok, as discussed earlier, I am removing any mentions of the shadow database.